pkg/processor/controlcommunication/controlcommunication_test.go

```diff
@@ -64,11 +65,89 @@ func (suite *ControlCommunicationTestSuite) TestSubscribeUnsubscribe() {
 	suite.Require().NoError(err)
 
 	// make sure the channel is unsubscribed
-	suite.Require().Len(suite.broker.Consumers, 1)
+	suite.Require().Len(suite.broker.Consumers, len(getAllControlMessageKinds()))
 	suite.Require().Len(suite.broker.Consumers[0].channels, 1)
 	suite.Require().Equal(suite.broker.Consumers[0].channels[0], controlMessageChannel2)
 }
 
+func (suite *ControlCommunicationTestSuite) TestSendMessage() {
+	controlMessageDrain1 := make(chan *ControlMessage)
+	controlMessageDrain2 := make(chan *ControlMessage)
+	defer close(controlMessageDrain1)
+	defer close(controlMessageDrain2)
+
+	drainDoneConsumer, err := suite.broker.getConsumer(DrainDoneMessageKind)
+	suite.Require().NoError(err)
+
+	for _, controlMessageChan := range []chan *ControlMessage{controlMessageDrain1, controlMessageDrain2} {
+		err = suite.broker.Subscribe(DrainDoneMessageKind, controlMessageChan)
+		suite.Require().NoError(err)
+	}
+
+	defer func() {
+		err := suite.broker.Unsubscribe(StreamMessageAckKind, controlMessageDrain1)
+		suite.Require().NoError(err)
+		err = suite.broker.Unsubscribe(StreamMessageAckKind, controlMessageDrain2)
+		suite.Require().NoError(err)
+	}()
+
+	// send DrainDone message
+	suite.Require().Len(drainDoneConsumer.channels, 2)
+	err = suite.broker.SendToConsumers(&ControlMessage{Kind: DrainDoneMessageKind})
+	suite.Require().NoError(err)
+	suite.Require().Len(drainDoneConsumer.channels, 0)
+
+	message := <-controlMessageDrain1
+	suite.Require().Equal(message.Kind, DrainDoneMessageKind)
+	message = <-controlMessageDrain2
+	suite.Require().Equal(message.Kind, DrainDoneMessageKind)
+
+}
+
+func (suite *ControlCommunicationTestSuite) TestBroadcastMessage() {
+	controlMessageExplicitAck1 := make(chan *ControlMessage)
+	controlMessageExplicitAck2 := make(chan *ControlMessage)
+	defer close(controlMessageExplicitAck1)
+	defer close(controlMessageExplicitAck2)
```

Comment:
why not in a single deferred function?

Reply:
@TomerShor I just made like this because of less lines, but it doesn't really matter